### PR TITLE
chore: refactor fuzz functions into share test package

### PR DESCRIPTION
**Pull request by justinsb**

### CODE REVIEW

File: pkg/test/fuzz/generate.go [hunk 1]
```diff
@@ -0,0 +1,316 @@
...
+		case protoreflect.MessageKind:
+			fieldVal := msg.Mutable(field)
+			fillWithRandom0(t, randStream, fieldVal.Message())
+
+		case protoreflect.BoolKind:
+			msg.Set(field, protoreflect.ValueOfBool(randStream.Intn(2) == 1))
+
+		case protoreflect.DoubleKind:
+			msg.Set(field, protoreflect.ValueOfFloat64(randStream.NormFloat64()))
+		case protoreflect.Int32Kind:
```
Comment (yuwenma) [inline pkg/test/fuzz/generate.go:117]: I feel like some resource may also use `UInt32Kind` and `UInt64Kind`

File: pkg/test/fuzz/generate.go [hunk 2]
```diff
@@ -0,0 +1,316 @@
...
+			msg.Set(field, protoreflect.ValueOf(val.Number()))
+		default:
+			t.Fatalf("unhandled field kind %v: %v", field.Kind(), field)
+		}
+	}
+}
+
+func randomString(randStream *rand.Rand) string {
+	// TODO: This is not a good random string!
+	return fmt.Sprintf("%x", randStream.Int63())
```
Comment (yuwenma) [inline pkg/test/fuzz/generate.go:137]: (but this is really nit!)

**Review by yuwenma — COMMENTED** [review]
/lgtm
/approve

Thank you!

File: pkg/test/fuzz/generate.go (hunk 1 above)
Comment (justinsb) [reply pkg/test/fuzz/generate.go:117]: Yes, but we will t.Fatalf when we hit one :-)

File: pkg/test/fuzz/generate.go [hunk 3]
```diff
@@ -0,0 +1,316 @@
...
+			fieldVal := msg.Mutable(field)
+			fillWithRandom0(t, randStream, fieldVal.Message())
+
+		case protoreflect.BoolKind:
+			msg.Set(field, protoreflect.ValueOfBool(randStream.Intn(2) == 1))
+
+		case protoreflect.DoubleKind:
+			msg.Set(field, protoreflect.ValueOfFloat64(randStream.NormFloat64()))
+		case protoreflect.Int32Kind:
+			msg.Set(field, protoreflect.ValueOfInt32(randStream.Int31()))
```
[the remark this answers is not in the document]
Comment (justinsb) [reply pkg/test/fuzz/generate.go:118]: Good call - do you want to fix it? :-)

**Pull request merged**